main.cpp

```diff
@@ -0,0 +1,297 @@
+#include <iostream>

```

Comment:
Do add a brief comment here explaining what exactly your implementation does.

Reply:
A structure with 2 pointers left and right which acts as left and right threads pointing to inorder predecessor and inorder successor respectively.Class is created with necessary functions and the members. Function create will create new nodes and constructs the tree by following structure and order property of binary tree. Display function which performs the inorder traversal of tree.Delete function which will delete the key node inputted,if the search for that key is successful. Deletion is concerned with 3 kinds-deleting a node which has 2 child,node with only left/right child,node with no child.